### PR TITLE
Add flags to common schema payload

### PR DESCRIPTION
**Pull request by guperrot**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/OneCollectorChannelListenerTest.java [hunk 1]
```diff
@@ -157,15 +165,16 @@ public void enqueueConvertedLogs() {
...
-        when(log4.getIKey()).thenReturn("t2");
+        log4.setExt(ext4);
         when(logSerializer.toCommonSchemaLog(any(Log.class))).thenReturn(Collections.singletonList(log4));
-        listener.onPreparedLog(originalLog, TEST_GROUP, DEFAULT_FLAGS);
+        listener.onPreparedLog(originalLog, TEST_GROUP, PERSISTENCE_NORMAL);
 
-        /* Verify reset of epoch/seq. */
+        /* Verify flags and reset of epoch/seq. */
+        assertEquals(new Long(PERSISTENCE_NORMAL), log4.getFlags());
         assertEquals((Long) 1L, log4.getExt().getSdk().getSeq());
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/OneCollectorChannelListenerTest.java]: Let's use only one way for casting numbers in asserts like this. For example
```suggestion
         assertEquals(Long.valueOf(1), log4.getExt().getSdk().getSeq());
```

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/OneCollectorChannelListenerTest.java [hunk 2]
```diff
@@ -92,18 +95,18 @@ public void enqueueConvertedLogs() {
         when(originalLog.getTransmissionTargetTokens()).thenReturn(new HashSet<>(Collections.singletonList("t1")));
 
         /* Mock a log. */
-        CommonSchemaLog log1 = mock(CommonSchemaLog.class);
```
Comment (jwallra) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/OneCollectorChannelListenerTest.java]: I know this is not necessarily in scope for this PR, but this test is doing a lot.  I'd like to see it split up.

Comment (guperrot) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/OneCollectorChannelListenerTest.java]: Not sure how to split as we need to test reset the sequence of epoch and seq in the same test.

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/OneCollectorChannelListenerTest.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/OneCollectorChannelListenerTest.java]: Ok.

**Pull request merged**